code/modules/ruins/ghost_bar.dm

```diff
@@ -15,16 +15,17 @@
 	if(alert(user, "Would you like to use one of your saved characters in your character creator?",, "Yes", "No") == "Yes")
 		var/list/our_characters_names = list()
 		var/list/our_character_saves = list()
-		for(var/datum/character_save/saves in user.client.prefs.character_saves)
-			our_characters_names += saves.real_name
-			our_character_saves += list(saves.real_name = saves)
+		for(var/index in 1 to length(user.client.prefs.character_saves))
+			var/datum/character_save/saves = user.client.prefs.character_saves[index]
+			our_characters_names += "[saves.real_name] (Slot #[index])"
+			our_character_saves += list("[saves.real_name] (Slot #[index])" = saves)
 
-		var/character_name = tgui_input_list(user, "Select a character", "Character selection", our_characters_names)
-		if(!character_name)
+		var/character_index = tgui_input_list(user, "Select a character", "Character selection", our_characters_names)
```

Comment:
wouldn't `character_name` still be the appropriate name here? maybe `character_slot`? either way `index` suggests an integer